File: Exo1.1.1.js
function getRandomInt(max) {
    return Math.floor(Math.random() * Math.floor(max));
}

var nombre_aleatoire = getRandomInt(101)
var message = "Devinez un nombre entre 1 et 100 "
var essai1 = 0

var comparaison = function(nombre,essai){
    if(essai < nombre){
	if(nombre-essai <= 5){
	    message="C'est un tout petit peu plus "
	}else if(nombre-essai < 10){
	    message="C'est un peu plus "
	}else{
	    message="C'est beaucoup plus "
	}
    }else if(essai > nombre){
	if(essai-nombre <= 5){
	    message="C'est un tout petit peu moins "
	}else if(essai-nombre < 10){
	    message="C'est un peu moins "
	}else{
	    message="C'est beaucoup moins "
	}
    }else{
	message="Vous avez gagne "
    }
    return nombre-essai
}
    
var readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

var jeu = function(nombre,essai){
    if(comparaison(nombre,essai) != 0){
	rl.question(message, (answer)=> {
	    essai1=answer
	    jeu(nombre,answer)
	});
    }else{
	console.log(message)
	rl.close();	
    }
}

rl.question(message, (answer) => {
    jeu(nombre_aleatoire,answer)
});

File: Exo1.1.1.py
# -*-coding:Latin-1 -*
import os # On importe le module os
from random import *

nombre_aleatoire = randrange(101)

print("Devinez un nombre entre 1 et 100")
essai = int(input())

def comparaison(nombre, essai) :
	if essai < nombre  :
		if nombre-essai <= 5 :
			print("C'est un tout petit peu plus")
			
		elif nombre-essai < 10 :
			print("C'est un peu plus")
			
		else :
			print("C'est beaucoup plus")
			
	elif essai > nombre:
		if essai-nombre <= 5 :
			print("C'est un tout petit peu moins")
			
		elif essai-nombre < 10 :
			print("C'est un peu moins")
			
		else :
			print("C'est beaucoup moins")
	else :
		print("Vous avez gagne")		
	return	nombre-essai		

while comparaison(nombre_aleatoire,essai) != 0 :
	essai = int(input())

os.system("pause")